sapi/cli/php_cli.c

```diff
@@ -1267,29 +1267,50 @@ int main(int argc, char *argv[])
 				break;
 			case 'd': {
 				/* define ini entries on command line */
-				int len = (int)strlen(php_optarg);
-				char *val;
+				size_t len = strlen(php_optarg);
+				char *val, *pini_entries;
 
 				if ((val = strchr(php_optarg, '='))) {
 					val++;
 					if (!isalnum(*val) && *val != '"' && *val != '\'' && *val != '\0') {
-						ini_entries = realloc(ini_entries, ini_entries_len + len + sizeof("\"\"\n\0"));
+						if ((ini_entries_len + len + sizeof("\"\"\n\0")) >= SIZE_MAX) {
```

Comment:
Generally all checks of the form `>= SIZE_MAX` are ineffective, because the overflow will already happen before the comparison.